framework/entityext/servicedef/services.xml

```diff
@@ -447,7 +447,7 @@ under the License.
         <auto-attributes include="pk" mode="IN" optional="false"/>
     </service>
     <service name="entitySyncPermissionCheck" engine="simple"
-             location="component://entityext/minilang/EntitySyncServices.xml" invoke="entitySyncPermissionCheck">
+             location="component://entityext/groovyScripts/EntitySyncServices.groovy" invoke="entitySyncPermissionCheck">
```

Comment:
IMO, this can be made simpler using:

<service name="entitySyncPermissionCheck" engine="simple"
             location="component://common/minilang/permission/CommonPermissionServices.xml" invoke="genericBasePermissionCheck">
        <implements service="permissionInterface"/>
        <attribute name="primaryPermission" type="String" mode="IN" optional="true" default-value="ENTITY_SYNC"/>
        <attribute name="altPermission" type="String" mode="IN" optional="true"/>
    </service>

After this, you didn't need a groovy method.

Reply:
Thanks for the suggestion, @surajkhurana. I will look into that.

Reply:
Hi @surajkhurana, I was inclined to agree about your suggested change reducing the amount of code in ofbiz and was about to raise a jira issue to refactor other similar permission check services to use the Common Permission Services.

I then realised that doing so would couple the various permission check services to the implementation of the genericBasePermissionCheck, in this case a minilang implementation.

If we later want to re-implement genericBasePermissionCheck using a different engine (groovy, java, etc) then we will need to update all the application specific permission check services that depend on it.

In this case I think it is better to have application specific permission check services depend on the genericBasePermissionCheck service-interface and not the service-implementation.